File: easy/1.TwoSum.py
# 1. Two Sum
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        '''
        Check each element, if we've not visited its other part
        (target - itself), store this element and its index;
        otherwise, we have an answer.
        '''
        seen = {}
        for i, a in enumerate(nums):
            prev_i = seen.get(target - a, None)
            if prev_i is None:
                seen[a] = i
            else:
                return [prev_i, i]

File: easy/1. Two Sum.rb
# 1. Two Sum
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    seen = Hash.new
    nums.each_with_index do |val, index|
        prev_index = seen[target - val]
        return [prev_index, index] if prev_index.nil?
        seen[val] = index
    end
end